Sources/Bow/Transformers/EnvT.swift

```diff
@@ -0,0 +1,115 @@
+public final class ForEnvT {}
+public final class EnvTPartial<E, W>: Kind2<ForEnvT, E, W> {}
+public typealias EnvTOf<E, W, A> = Kind<EnvTPartial<E, W>, A>
+
+public typealias ForEnv = ForEnvT
+public typealias EnvPartial<E> = EnvTPartial<E, ForId>
+public typealias Env<E, A> = EnvT<E, ForId, A>
+
+public final class EnvT<E, W, A>: EnvTOf<E, W, A> {
+    fileprivate let e: E
+    fileprivate let wa: Kind<W, A>
+    
+    public static func fix(_ value: EnvTOf<E, W, A>) -> EnvT<E, W, A> {
+        value as! EnvT<E, W, A>
+    }
+    
+    public init(_ e: E, _ wa: Kind<W, A>) {
+        self.e = e
+        self.wa = wa
+    }
+    
+    public init(_ pair: (E, Kind<W, A>)) {
+        self.e = pair.0
+        self.wa = pair.1
+    }
+    
+    public func runT() -> (E, Kind<W, A>) {
+        (e, wa)
+    }
+    
+    public func local<EE>(_ f: @escaping (E) -> EE) -> EnvT<EE, W, A> {
+        EnvT<EE, W, A>(f(e), wa)
+    }
+}
+
+public postfix func ^<E, W, A>(_ value: EnvTOf<E, W, A>) -> EnvT<E, W, A> {
+    EnvT.fix(value)
+}
+
+// MARK: Syntax for Env
+
+public extension EnvT where W == ForId {
+    convenience init(_ e: E, _ a: A) {
+        self.init(e, Id(a))
+    }
+    
+    func run() -> (E, A) {
+        let (e, wa) = runT()
+        return (e, wa^.value)
+    }
+}
+
+// MARK: Instance of `Invariant` for `EnvT`
+
+extension EnvTPartial: Invariant where W: Functor {}
+
+// MARK: Instance of `Functor` for `EnvT`
+
+extension EnvTPartial: Functor where W: Functor {
+    public static func map<A, B>(_ fa: EnvTOf<E, W, A>, _ f: @escaping (A) -> B) -> EnvTOf<E, W, B> {
+        EnvT(fa^.e, fa^.wa.map(f))
+    }
+}
+
+// MARK: Instance of `Applicative` for `EnvT`
+
+extension EnvTPartial: Applicative where W: Applicative, E: Monoid {
+    public static func pure<A>(_ a: A) -> Kind<EnvTPartial<E, W>, A> {
+        EnvT(E.empty(), W.pure(a))
+    }
+    
+    public static func ap<A, B>(_ ff: EnvTOf<E, W, (A) -> B>, _ fa: EnvTOf<E, W, A>) -> EnvTOf<E, W, B> {
+        EnvT(ff^.e.combine(fa^.e), ff^.wa.ap(fa^.wa))
+    }
+}
+
+// MARK: Instance of `Comonad` for `EnvT`
+
+extension EnvTPartial: Comonad where W: Comonad {
+    public static func coflatMap<A, B>(_ fa: EnvTOf<E, W, A>, _ f: @escaping (EnvTOf<E, W, A>) -> B) -> EnvTOf<E, W, B> {
+        EnvT(fa^.e, fa^.wa.coflatMap { a in f(EnvT(fa^.e, a)) })
+    }
+    
+    public static func extract<A>(_ fa: Kind<EnvTPartial<E, W>, A>) -> A {
+        fa^.wa.extract()
+    }
+}
+
+// MARK: Instance of `ComonadEnv` for `EnvT`
+
+extension EnvTPartial: ComonadEnv where W: Comonad {
+    public static func ask<A>(_ wa: Kind<EnvTPartial<E, W>, A>) -> E {
+        wa^.e
+    }
+}
+
```

Comment:
Should we provide an instance of ComonadTrans to EnvT?

```suggestion

// MARK: Instance of `ComonadTrans` for `EnvT`

extension EnvTPartial: ComonadTrans {
    public static func lower<W: Comonad, A>(_ twa: Kind<EnvTPartial, Kind<W, A>>) -> Kind<W, A> {
        twa.lower()
    }
}
```